Race condition in `make install`

Hi,

I've been working on making a weechat-matrix package for gentoo and on my journey I've found that occasionally the install fails with:

```
make -j5 DESTDIR=/var/tmp/portage/net-irc/weechat-matrix-0.2.0/image PREFIX=/usr/lib64/weechat install 
install -d /var/tmp/portage/net-irc/weechat-matrix-0.2.0/image/usr/lib64/weechat/python/matrix
install -m644 matrix/buffer.py /var/tmp/portage/net-irc/weechat-matrix-0.2.0/image/usr/lib64/weechat/python/matrix/buffer.py
install -m644 matrix/utf.py /var/tmp/portage/net-irc/weechat-matrix-0.2.0/image/usr/lib64/weechat/python/matrix/utf.py
install -m644 matrix/colors.py /var/tmp/portage/net-irc/weechat-matrix-0.2.0/image/usr/lib64/weechat/python/matrix/colors.py
install -m644 matrix/server.py /var/tmp/portage/net-irc/weechat-matrix-0.2.0/image/usr/lib64/weechat/python/matrix/server.py
/usr/bin/install: cannot create regular file '/var/tmp/portage/net-irc/weechat-matrix-0.2.0/image/usr/lib64/weechat/python/matrix/utf.py': No such file or directory
install -m644 matrix/bar_items.py /var/tmp/portage/net-irc/weechat-matrix-0.2.0/image/usr/lib64/weechat/python/matrix/bar_items.py
make: *** [Makefile:28: /var/tmp/portage/net-irc/weechat-matrix-0.2.0/image/usr/lib64/weechat/python/matrix/utf.py] Error 1
make: *** Waiting for unfinished jobs....
```

Now I believe the important part is the `-j5` where the file is being attempted to copy into a directory that doesn't exist yet.

I think I have a PR with a fix incoming

Just following up, this reproduces it:
```
$ while true; do rm -rf dest/home && make -j5 DESTDIR=$PWD/dest install > /dev/null; done
install: cannot create regular file '/home/msirabella/Documents/projects/featurefork/weechat-matrix/dest/home/msirabella/.weechat/python/matrix/buffer.py': No such file or directory
make: *** [Makefile:28: /home/msirabella/Documents/projects/featurefork/weechat-matrix/dest/home/msirabella/.weechat/python/matrix/buffer.py] Error 1
make: *** Waiting for unfinished jobs....
^Cmake: *** [Makefile:19: install-dir] Interrupt
```

--- a/Makefile
+++ b/Makefile
@@ -11,12 +11,12 @@ lib := $(patsubst matrix/%.py, $(DESTDIR)$(PREFIX)/python/matrix/%.py, \
 help:
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-install: install-dir install-lib ## Install the plugin to $(DESTDIR)/$(PREFIX)
+install: install-lib | $(DESTDIR)$(PREFIX)/python/matrix/ ## Install the plugin to $(DESTDIR)/$(PREFIX)
 	install -m644 main.py $(DESTDIR)$(PREFIX)/python/matrix.py
 
 install-lib: $(lib)
-install-dir:
-	install -d $(DESTDIR)$(PREFIX)/python/matrix
+$(DESTDIR)$(PREFIX)/python/matrix/:
+	install -d $@
 
 uninstall: ## Uninstall the plugin from $(PREFIX)
 	rm $(DESTDIR)$(PREFIX)/python/matrix.py $(DESTDIR)$(PREFIX)/python/matrix/*
@@ -24,7 +24,7 @@ uninstall: ## Uninstall the plugin from $(PREFIX)
 
 phony:
 
-$(DESTDIR)$(PREFIX)/python/matrix/%.py: matrix/%.py phony
+$(DESTDIR)$(PREFIX)/python/matrix/%.py: matrix/%.py phony | $(DESTDIR)$(PREFIX)/python/matrix/
 	install -m644 $< $@
 
 test: ## Run automated tests

--- a/Makefile
+++ b/Makefile
@@ -4,27 +4,29 @@ WEECHAT_HOME ?= $(HOME)/.weechat
 PREFIX ?= $(WEECHAT_HOME)
 PYTHON ?= python
 
-lib := $(patsubst matrix/%.py, $(DESTDIR)$(PREFIX)/python/matrix/%.py, \
+INSTALLDIR := $(DESTDIR)$(PREFIX)/python/matrix
+
+lib := $(patsubst matrix/%.py, $(INSTALLDIR)/%.py, \
 	 $(wildcard matrix/*.py))
 
 .PHONY: help
 help:
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-install: install-lib | $(DESTDIR)$(PREFIX)/python/matrix/ ## Install the plugin to $(DESTDIR)/$(PREFIX)
+install: install-lib | $(INSTALLDIR) ## Install the plugin to $(DESTDIR)/$(PREFIX)
 	install -m644 main.py $(DESTDIR)$(PREFIX)/python/matrix.py
 
 install-lib: $(lib)
-$(DESTDIR)$(PREFIX)/python/matrix/:
+$(INSTALLDIR):
 	install -d $@
 
 uninstall: ## Uninstall the plugin from $(PREFIX)
-	rm $(DESTDIR)$(PREFIX)/python/matrix.py $(DESTDIR)$(PREFIX)/python/matrix/*
-	rmdir $(DESTDIR)$(PREFIX)/python/matrix
+	rm $(DESTDIR)$(PREFIX)/python/matrix.py $(INSTALLDIR)/*
+	rmdir $(INSTALLDIR)
 
 phony:
 
-$(DESTDIR)$(PREFIX)/python/matrix/%.py: matrix/%.py phony | $(DESTDIR)$(PREFIX)/python/matrix/
+$(INSTALLDIR)/%.py: matrix/%.py phony | $(INSTALLDIR)
 	install -m644 $< $@
 
 test: ## Run automated tests

